packages/web/src/pages/search-page-v2/SearchPageV2.tsx

```diff
@@ -0,0 +1,178 @@
+import MobilePageContainer from 'components/mobile-page-container/MobilePageContainer'
+import NavContext, {
+  CenterPreset,
+  LeftPreset,
+  RightPreset
+} from 'components/nav/mobile/NavContext'
+import Page from 'components/page/Page'
+import { useMedia } from 'hooks/useMedia'
+
+import { SearchCatalogTile } from './SearchCatalogTile'
+
+type Filter =
+  | 'genre'
+  | 'mood'
+  | 'key'
+  | 'bpm'
+  | 'isPremium'
+  | 'hasDownloads'
+  | 'isVerified'
+
+type Category = {
+  filters: Filter[]
+  icon?: IconComponent
+}
+
+const categories = {
+  all: { filters: [] },
+  profiles: { icon: IconUser, filters: ['genre', 'isVerified'] },
+  tracks: {
+    icon: IconNote,
+    filters: ['genre', 'mood', 'key', 'bpm', 'isPremium', 'hasDownloads']
+  },
+  albums: { icon: IconAlbum, filters: ['genre', 'mood'] },
+  playlists: { icon: IconPlaylists, filters: ['genre', 'mood'] }
+} satisfies Record<string, Category>
+
+type CategoryKey = keyof typeof categories
+
+type SearchHeaderProps = {
+  category?: CategoryKey
+  setCategory: (category: CategoryKey) => void
+  title: string
+  query: Maybe<string>
+}
+
+const SearchHeader = (props: SearchHeaderProps) => {
+  const { category: categoryKey = 'all', setCategory, query, title } = props
+
+  const { isMobile } = useMedia()
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const value = e.target.value
+      setCategory(value as CategoryKey)
+    },
+    [setCategory]
+  )
+
+  const CategoryRadioGroup = (
+    <RadioGroup
+      direction='row'
+      gap='s'
+      aria-label={'Select search category'}
+      name='searchcategory'
+      value={categoryKey}
+      onChange={handleChange}
+    >
+      {Object.entries(categories)
+        .filter(([key]) => !isMobile || key !== 'all')
+        .map(([key, category]) => (
+          <SelectablePill
+            aria-label={`${key} search category`}
+            icon={(category as Category).icon}
+            key={key}
+            label={capitalize(key)}
+            size='large'
+            type='radio'
+            value={key}
+            checked={key === categoryKey}
+          />
+        ))}
+    </RadioGroup>
+  )
+
+  return isMobile ? (
+    <Flex p='s' css={{ overflow: 'scroll' }}>
+      {CategoryRadioGroup}
+    </Flex>
+  ) : (
+    <Header
+      {...props}
+      primary={title}
+      secondary={
+        query ? (
+          <Flex ml='l'>
+            <Text variant='heading' strength='weak'>
+              &#8220;{query}&#8221;
+            </Text>
+          </Flex>
+        ) : null
+      }
+      rightDecorator={CategoryRadioGroup}
+      variant='main'
+    />
+  )
+}
+
+export const SearchPageV2 = () => {
+  const { isMobile } = useMedia()
+
+  const { category, query } = useParams<{
+    category: CategoryKey
+    query: string
+  }>()
+  const { history } = useHistoryContext()
+
+  // Set nav header
+  const { setLeft, setCenter, setRight } = useContext(NavContext)!
```

Comment:
ideally this is built into "page" but not something you need to change

Reply:
I agree, I don't like this current setup